static/app/views/releases/releaseBubbles/useReleaseBubbles.tsx

```diff
@@ -362,13 +363,55 @@ export function useReleaseBubbles({
     : Date.now();
   const chartRef = useRef<ReactEchartsRef | null>(null);
   const hasReleaseBubbles = organization.features.includes('release-bubbles-ui');
-  const handleChartRef = useCallback((e: ReactEchartsRef | null) => {
-    chartRef.current = e;
-
-    if (e?.getEchartsInstance) {
-      createReleaseBubbleHighlighter(e.getEchartsInstance());
-    }
-  }, []);
+  const totalBubblePaddingY = bubblePadding * 2;
+  const releaseBubbleXAxis = useMemo(
+    () => ({
+      // configure `axisLine` and `offset` to move axis line below 0 so that
+      // bubbles sit between bottom of the main chart and the axis line
+      axisLine: {onZero: false},
+      offset: bubbleSize + totalBubblePaddingY - 1,
+    }),
+    [bubbleSize, totalBubblePaddingY]
+  );
+  const releaseBubbleGrid = useMemo(
+    () => ({
+      // Moves bottom of grid "up" `bubbleSize` pixels so that bubbles are
+      // drawn below grid (but above x axis label)
+      bottom: bubbleSize + totalBubblePaddingY + 1,
+    }),
+    [bubbleSize, totalBubblePaddingY]
+  );
+
+  const handleChartRef = useCallback(
+    (e: ReactEchartsRef | null) => {
+      chartRef.current = e;
+
+      if (e?.getEchartsInstance) {
+        const echartsInstance = e.getEchartsInstance();
+        createReleaseBubbleHighlighter(echartsInstance, {
+          onLegendChange: (selected: boolean) => {
```

Comment:
I initially implemented this with setState in the hook, but that will cause a re-render and desync of states between echarts and the state in the hook.